nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/ShellUserGroupProvider.java

```diff
@@ -255,11 +256,19 @@ public void onConfigured(AuthorizerConfigurationContext configurationContext) th
             throw new AuthorizerCreationException(SYS_CHECK_ERROR, ioexc.getCause());
         }
 
+        // The next step is to add the user and group exclude regexes:
+        try {
+            excludeGroups = Pattern.compile(getProperty(configurationContext, EXCLUDE_GROUP_PROPERTY, ""));
+            excludeUsers = Pattern.compile(getProperty(configurationContext, EXCLUDE_USER_PROPERTY, ""));
+        } catch (final PatternSyntaxException e) {
+            throw new AuthorizerCreationException(e);
+        }
+
         // With our command set selected, and our system check passed, we can pull in the users and groups:
         refreshUsersAndGroups();
 
         // And finally, our last init step is to fire off the refresh thread:
-        scheduler.scheduleWithFixedDelay(this::refreshUsersAndGroups, initialDelay, fixedDelay, TimeUnit.SECONDS);
+        scheduler.scheduleWithFixedDelay(this::refreshUsersAndGroups, fixedDelay, fixedDelay, TimeUnit.SECONDS);
```

Comment:
`refreshUsersAndGroups` cannot allow any exceptions to escape. If one does, the scheduling will not continue. We typically have a `try {...} catch (Throwable t) {...}` to ensure this does not happen.

Reply:
For instance `LdapUserGroupProvider` [1]

[1] https://github.com/apache/nifi/blob/2491c51d34a4012dd6c9058f60f80ea2203e311b/nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/tenants/LdapUserGroupProvider.java#L395